### PR TITLE
Finish Routing, Text Field Component, Refactor Button, Start Game Page

### PR DESCRIPTION
**Pull request by Chadtech**

### CODE REVIEW

File: ui/src/lib.rs [hunk 1]
```diff
@@ -12,51 +12,53 @@ mod view;
 
 enum Model {
     PageNotFound,
-    Title(page::title::Model),
+    Title,
```
Comment (Chadtech) [inline ui/src/lib.rs:15]: I deleted `Title`s state since it didnt really have any state.

File: ui/src/page/start_game.rs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+}
+
+////////////////////////////////////////////////////////////////
+// UPDATE //
+////////////////////////////////////////////////////////////////
+
+pub fn update(msg: Msg, model: &mut Model) {
+    match msg {
+        Msg::StartClicked => {
+            // ????? Send http request!!!!
```
Comment (Chadtech) [inline ui/src/page/start_game.rs:51]: `!!!!!!!!!!!!!!!!` We can do it now!

File: ui/src/route.rs [hunk 3]
```diff
@@ -16,22 +16,33 @@ pub enum Route {
...
+            Route::StartGame => vec!["start-game"],
+            Route::Demo => vec!["demo"],
         }
     }
+    pub fn to_string(self) -> String {
+        let mut buf = String::new();
+
+        buf.push('/');
+
+        buf.push_str(&self.path().join("/"));
```
Comment (Chadtech) [inline ui/src/route.rs:31]: `buf.push('/')` ensures the url starts with a `/`, which turns out to be really critical to proper routing.
```
<a href="start-game"/>
```
reloads the whole page and restarts all the code
```
<a href="/start-game"/>
```
does not

File: ui/src/view/text_field.rs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+impl<T> TextField<T> {
+    pub fn view(self) -> Node<T>
+    where
+        T: Clone,
+        T: 'static,
+    {
+        let mut element: El<T> = El::empty(Tag::Custom(Cow::Borrowed("input")));
+
+        let on_input = self.on_input;
```
Comment (Chadtech) [inline ui/src/view/text_field.rs:40]: This is necessary. I dont know why. `self.on_input(event)` below did not work

File: ui/src/lib.rs (hunk 1 above)
Comment (VictorXLR) [reply ui/src/lib.rs:15]: Nice

**Review by VictorXLR — APPROVED** [review]

**Pull request merged**